### PR TITLE
test: cover the tools, job queue and Storybook client

### PR DESCRIPTION
**Pull request by freema**
Coverage was ~5% with 0 of 9 tools tested. This takes it to **87% statements / 85% branches** with 225 tests, and raises the `vitest.config.ts` thresholds to match so it ratchets rather than drifts.

### What is covered

| Area | Tests |
|---|---|
| `list_components`, `search_components`, `get_component_html` | filtering, pagination, compact/full output, v3 `stories` vs v5 `entries`, variant listing, async job vs sync |
| `get_component_dependencies`, `get_theme_info`, `get_external_css` | component detection heuristics, token bucketing, same-origin check, truncation |
| `job_status`, `job_list`, `job_cancel` | lifecycle, filters, unknown ids |
| `JobQueue` | queueing, failure capture, mid-flight cancellation, sync timeout, stats |
| `StorybookClient` | URL validation, `index.json` → `stories.json` fallback, caching, static-vs-browser decision, single browser launch under concurrency, retry after a failed launch |
| parsing/pagination/cache/error helpers | full behaviour incl. edge cases |

### Three defects the tests found, fixed here

**`parseCSSRules` mis-keyed every declaration after the first.** A single `([^:]+):\s*([^;]+)` scan across the rule body swallowed the separator, so `color: red; padding: 4px` parsed as `{ color: 'red', '; padding': '4px' }`. Now split on `;` first.

**`formatError` threw away the actual cause outside development.** The category templates are generic, so on their own they can be flatly wrong — a pagination failure (`Page 9 exceeds total pages (3)`) was reported as *"Unable to connect to Storybook"*. The caller here is usually a model, which then goes and debugs the wrong system. The real message is now always appended on a `Details:` line.

**`get_theme_info` filed unit-valued tokens as spacing regardless of name.** `isSpacingValue(value)` was evaluated inside the spacing branch, ahead of the breakpoint, shadow and radius checks, so `--breakpoint-md: 960px` and `--radius-sm: 2px` both landed in `spacing`. Worse, the "no breakpoints found" fallback then populated the response with hardcoded defaults while the design system's real breakpoints sat in the wrong bucket. Name-based rules now run first; value heuristics are the fallback.

### Still uncovered

`src/index.ts` (stdio wiring) and `puppeteer-client.ts` (needs a real browser) — both belong in an integration suite rather than here.

One thing left alone deliberately: `get_theme_info` still substitutes hardcoded breakpoints when a design system defines none. With the ordering fix real breakpoints are now found, but a caller still cannot tell the invented ones from real values. Worth a separate decision.

### CODE REVIEW

File: src/utils/error-formatter.ts [hunk 1]
```diff
@@ -83,10 +83,16 @@ export function formatError(
...
-  // Add debug information in development
-  if (includeDebug && originalError) {
-    const debugInfo = typeof originalError === 'string' ? originalError : originalError.message;
-    message += `\nDebug: ${debugInfo}`;
+  // Always surface what actually went wrong. The category templates are
+  // generic by design, so on their own they can be flatly wrong: a pagination
+  // error ("Page 9 exceeds total pages (3)") fell through to the connection
+  // template and told the caller Storybook was unreachable. The caller here is
+  // usually a model, which then goes and debugs the wrong thing.
+  if (originalError) {
```
Comment (freema) [inline src/utils/error-formatter.ts:91]: **[SUGGESTION]** The underlying error message (`originalError`) is now always appended to the caller-facing message, whereas before it was only included when `includeDebug` (dev/debug mode) was true. This is the intended fix (confirmed by the accompanying tests/comment) since the previous behavior hid the real cause from callers, but it does mean any detail in `originalError.message` — including from lower-level libraries (undici/Puppeteer) — is now unconditionally surfaced in production responses.

> Suggestion: No change required given the tool's local/dev-server context (Storybook fetch/par:se errors), but worth double-checking that no call site ever passes an error whose message could contain sensitive data (e.g. auth headers embedded in a URL) before relying on this in other contexts.

**Review by freema — COMMENTED** [review]
## CodeForge Review

**Verdict:** approve | **Score:** 9/10

Well-scoped PR that fixes three real bugs (theme token miscategorization, a regex that mangled every CSS declaration after the first, and error messages that hid the actual cause behind a generic template) and backs each fix with clear regression tests. Also adds a large, high-quality unit test suite and raises the coverage thresholds to match, replacing gates that were previously unenforced. No correctness, security, or performance problems found in the diff.

---
*Reviewed by [CodeForge](https://github.com/freema/codeforge)*

**Review by freema — COMMENTED** [review]
## CodeForge Review

**Verdict:** approve | **Score:** 9/10

Three targeted bug fixes (theme token categorization order, CSS declaration parsing, and error message detail surfacing) plus a large, high-quality regression/unit test suite that specifically exercises each fixed bug and raises coverage thresholds to match reality. No correctness, security, or performance issues found in the diff.

---
*Reviewed by [CodeForge](https://github.com/freema/codeforge)*
